From: Simon McVittie <smcv@collabora.com>
Date: Mon, 18 Jan 2021 18:07:38 +0000
Subject: dir: Pass environment via bwrap --setenv when running apply_extra

This means we can systematically pass the environment variables
through bwrap(1), even if it is setuid and thus is filtering out
security-sensitive environment variables. bwrap ends up being
run with an empty environment instead.

As with the previous commit, this regressed while fixing CVE-2021-21261.

Fixes: 6d1773d2 "run: Convert all environment variables into bwrap arguments"
Bug: https://github.com/flatpak/flatpak/issues/4080
Signed-off-by: Simon McVittie <smcv@collabora.com>
Applied-upstream: 1.10.1, commit:fb473cad801c6b61706353256cab32330557374a
---
 common/flatpak-dir.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common/flatpak-dir.c b/common/flatpak-dir.c
index 40a2e58..f79ebcd 100644
--- a/common/flatpak-dir.c
+++ b/common/flatpak-dir.c
@@ -6799,6 +6799,8 @@ apply_extra_data (FlatpakDir   *self,
                                          app_context, NULL, NULL, cancellable, error))
     return FALSE;
 
+  flatpak_bwrap_envp_to_args (bwrap);
+
   flatpak_bwrap_add_arg (bwrap, "/app/bin/apply_extra");
 
   flatpak_bwrap_finish (bwrap);
